modules/packages/memorybuffer.chpl

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2004-2016 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+  code module to freeze/thaw data types into byte arrays
+ */
+module memorybuffer {
+
+  record MemoryBuffer {
+    type eltype;
+    const memstyle = new iostyle(binary=1);
+    param byteorder = iokind.native;
+    var buffd:file;
+    var bufsize:int;
+
+    proc MemoryBuffer(type eltype, data_item:eltype) {
```

Comment:
@ct-clmsn This buffer seems to only support serializing a single object via `file.writer()`, so it could not necessarily be used to serialize multiple objects not already in an array or of different types. Is that correct?